controllers/infra/zone/zone_controller.go

```diff
@@ -123,48 +123,52 @@ func (r *Reconciler) ReconcileDelete(
 	ctx context.Context,
 	obj *topologyv1.Zone) (ctrl.Result, error) {
 
-	if controllerutil.ContainsFinalizer(obj, Finalizer) {
-		if val := obj.Spec.ManagedVMs.FolderMoID; val != "" {
-			if err := watcher.Remove(
-				ctx,
-				vimtypes.ManagedObjectReference{
-					Type:  "Folder",
-					Value: val,
-				},
-				fmt.Sprintf("%s/%s", obj.Namespace, obj.Name)); err != nil {
-
-				if !errors.Is(err, watcher.ErrAsyncSignalDisabled) {
-					return ctrl.Result{}, err
-				}
+	if val := obj.Spec.ManagedVMs.FolderMoID; val != "" {
+		if err := watcher.Remove(
+			ctx,
+			vimtypes.ManagedObjectReference{
+				Type:  "Folder",
+				Value: val,
+			},
+			fmt.Sprintf("%s/%s", obj.Namespace, obj.Name)); err != nil {
+
+			if !errors.Is(err, watcher.ErrAsyncSignalDisabled) {
+				// We don't ignore watcher.ErrNoWatcher here to interlock with the vm watcher
+				// service that is in the process of restarting the watcher. This does mean
+				// that if watcher cannot start like because of invalid VC creds the finalizer
+				// won't be removed.
```

Comment:
Sure, the finalizer won't be removed, but the zone will get requeued. 

We could have the same discussion about the inability to delete vSphere VMs if their Kube object has a deletion timestamp and the creds are bad.